### PR TITLE
feat: setup Google Analytics tags on UI

### LINKED ISSUE #311: Setup Google Analytics tags on UI
Setup Google Analytics tags on UI to track usage
Requirements:
- No DEV and local environment tags
- "?" QA environment tags

Review for environment setup https://support.google.com/tagmanager/answer/6311518?hl=en

### PR DESCRIPTION
**Pull request by davidgamez**
**Summary:**

Add Google Analytics to the pageview event per environment:
- Dev/local/test/cypress: no GA tag added to the page
- QA: GA tag added reporting to qa.mobilitydatabase.org site
- PROD: GA tag added reporting to mobilitydatabase.org site

Closes #311 

**Expected behavior:** 

Google Analytics requests are visible from GA dashboard.

**Testing tips:**

GA requests are sent to the analytics dashboard
- Deploy to QA from this branch
- Open https://qa.mobilitydatabase.org
- Open developer console
- Refresh the page and verify that the request with URL _https://www.googletagmanager.com/gtag/js?id={GA_ID}_ URL is visible on the network tab
- Go to the Google Analytics dashboard and verify that requests are generating statistics

Local 
- Start the web-app locally, `yarn start:dev`
- Open developer console
- Refresh the page and verify that the request with _https://www.googletagmanager.com/gtag/js?id={GA_ID}_ is URL **not** visible on the network tab

QA GA report:
Screenshot 2024-02-28 at 1 19 50 PM

- [x] Run the unit tests with `./scripts/api-tests.sh` to make sure you didn't break anything
- [x] Add or update any needed documentation to the repo
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [x] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

File: .github/workflows/web-app-deployer.yml [hunk 1]
```diff
@@ -16,6 +16,12 @@ on:
         description: Host name to be used in preview deployment
         type: string
         required: false
+      REACT_APP_GOOGLE_ANALYTICS_ID:
+        description: Google Analytics ID
+        type: string
+        # defaulting to a placeholder to keep the GA ID empty if parameter is not passed
+        default: '{{REACT_APP_GOOGLE_ANALYTICS_ID}}'
```
Comment (jcpitre) [inline .github/workflows/web-app-deployer.yml:23]: This looks wrong. Where's the $

Comment (davidgamez) [reply .github/workflows/web-app-deployer.yml:23]: This is the placeholder of the property file, [here](https://github.com/MobilityData/mobility-feed-api/blob/bebf0208bc8d1a4e1f4de72be1365bb35b5ad42c/web-app/src/.env.rename_me#L9). So the value will be kept as the placeholder, this makes the environment to be empty at runtime, implementation [here](https://github.com/MobilityData/mobility-feed-api/blob/bebf0208bc8d1a4e1f4de72be1365bb35b5ad42c/web-app/src/app/utils/config.ts#L7)

Comment (davidgamez) [reply .github/workflows/web-app-deployer.yml:23]: Keeping the placeholder, let's keep local development and testing and ignore the value. The pattern `key={{key}}` is consistently use in the configuration file. Having a "NO_VALUE" is also an option, but requires that the develop[per actually add the value to the property file. It's easier to keep the property file unmodified and get empty string.

**Review by jcpitre — APPROVED** [review]
LGTM!

**Pull request merged**